src/main/java/com/google/api/generator/gapic/composer/ResourceNameHelperClassComposer.java

```diff
@@ -1140,6 +1145,113 @@ private static MethodDefinition createToStringMethod(
         .build();
   }
 
+  private static MethodDefinition createEqualsMethod(
+      ResourceName resourceName, List<List<String>> tokenHierarchies, Map<String, TypeNode> types) {
+    Variable oVariable = Variable.builder().setType(TypeNode.OBJECT).setName("o").build();
+    VariableExpr argVarExprDecl =
+        VariableExpr.builder().setIsDecl(true).setVariable(oVariable).build();
+    VariableExpr argVarExprNotDecl =
+        VariableExpr.builder().setIsDecl(false).setVariable(oVariable).build();
+    TypeNode thisClassType = types.get(getThisClassName(resourceName));
+    ValueExpr thisValueExpr = ValueExpr.withValue(ThisObjectValue.withType(thisClassType));
+    ValueExpr trueValueExpr =
+        ValueExpr.withValue(
+            PrimitiveValue.builder().setType(TypeNode.BOOLEAN).setValue("true").build());
+    ValueExpr falseValueExpr =
+        ValueExpr.withValue(
+            PrimitiveValue.builder().setType(TypeNode.BOOLEAN).setValue("false").build());
+
+    ReturnExpr returnTrueExpr = ReturnExpr.withExpr(trueValueExpr);
+    RelationalOperationExpr oEqualsThisExpr =
+        RelationalOperationExpr.equalToWithExprs(argVarExprNotDecl, thisValueExpr);
+    RelationalOperationExpr oNotEqualsNullExpr =
+        RelationalOperationExpr.notEqualToWithExprs(
+            argVarExprNotDecl, ValueExpr.withValue(NullObjectValue.create()));
+    MethodInvocationExpr getClassMethodInvocationExpr =
+        MethodInvocationExpr.builder().setMethodName("getClass").build();
+    RelationalOperationExpr getClassEqualsExpr =
+        RelationalOperationExpr.equalToWithExprs(
+            getClassMethodInvocationExpr,
+            getClassMethodInvocationExpr
+                .toBuilder()
+                .setExprReferenceExpr(argVarExprNotDecl)
+                .build());
+    LogicalOperationExpr orLogicalExpr =
+        LogicalOperationExpr.logicalOrWithExprs(oNotEqualsNullExpr, getClassEqualsExpr);
+
+    Variable thatVariable = Variable.builder().setName("that").setType(thisClassType).build();
+    VariableExpr thatVariableExpr =
+        VariableExpr.builder().setIsDecl(true).setVariable(thatVariable).build();
```

Comment:
Could we not set this property, and instead use `toBuilder().setIsDecl(true).toBuild()` where needed?

Reply:
sg